### PR TITLE
chore: fix tests using parent account

### PR DESCRIPTION
**Pull request by sobolk**
#### Description of changes

This PR:
- fixes `useChildAccountCredentials` - the `exit 0` was causing premature job termination on circle ci, i.e. **tests that used parent account were not run !!!**
- moves some tests from parent account to child accounts
- tweaks re-enabled tests to make them passing (or almost passing)

#### Description of how you validated changes

E2E tests.

The `searchable-migration` test keeps failing due to throttling. It will require follow up work that's not in scope of this PR.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by awsluja — DISMISSED** [review]
lgtm w/ passing e2e

File: packages/amplify-e2e-tests/src/__tests__/__snapshots__/api_10.test.ts.snap [hunk 1]
```diff
@@ -15,6 +15,9 @@ $util.unauthorized()
   #set( $currentClaim1 = $util.defaultIfNull($ctx.identity.claims.get(\\"username\\"), $util.defaultIfNull($ctx.identity.claims.get(\\"cognito:username\\"), null)) )
   #if( !$util.isNull($ownerClaim0) && !$util.isNull($currentClaim1) )
     #set( $ownerClaim0 = \\"$ownerClaim0::$currentClaim1\\" )
+    #if( $isAuthorized && $util.isNull($ownerEntity0) && !$ctx.args.input.containsKey(\\"owner\\") )
```
Comment (jhockett) [inline packages/amplify-e2e-tests/src/__tests__/__snapshots__/api_10.test.ts.snap:18]: Any idea how this snapshot got out of sync?

**Review by jhockett — DISMISSED** [review]

File: packages/amplify-e2e-tests/src/__tests__/__snapshots__/api_10.test.ts.snap (hunk 1 above)
Comment (sobolk) [reply packages/amplify-e2e-tests/src/__tests__/__snapshots__/api_10.test.ts.snap:18]: of course, this test never run after the `useChildAccountCredentials` got broken. 
few weeks ago we were wondering why such change was caught by tests in data but not cli... now we know...

**Review by awsluja — APPROVED** [review]
looks good & thanks for splitting that test up

**Review by jhockett — DISMISSED** [review]

**Review by awsluja — APPROVED** [review]

**Pull request merged**